execution_workflows/APAtrap/conf/modules.config

```diff
@@ -0,0 +1,81 @@
+/*
+ * --------------------------------------------------
+ *  Config file for defining DSL2 per module options
+ * --------------------------------------------------
+ *
+ *   Available keys to override module options:
+ *       args           = Additional arguments appended to command in module.
+ *       args2          = Second set of arguments appended to command in module (multi-tool modules).
+ *       publish_dir    = Directory to publish results.
+ *       publish_by_id  = Publish results in separate folders by meta.id value.
+ *       publish_files  = Groovy map where key = "file_ext" and value = "directory" to publish results for that file extension
+ *                        The value of "directory" is appended to the standard "publish_dir" path as defined above.
+ *                        If publish_files == null (unspecified)  - All files are published.
+ *                        If publish_files == false               - No files are published.
+ *       suffix         = File name suffix for output files.
+ *
+ */
+
+params {
+    workflow {
+        // if set to false, only identification and quantification outputs are produced
+        // if set to true, two samples need to be provided in samplesheet and only differential output is produced
+        run_differential = false
+
+        //Set the names of the files for each challenge
+        identification_out_suffix = 'identification_output_test.bed'
+        quantification_out_suffix = 'quantification_output_test.bed'
+        differential_out = 'apatrap_differential_output_test.tsv'
+    }
+    modules {
+        'check_samplesheet' {
+            publish_dir   = 'public_data'
+        }
+        'identify_distal_3utr' {
+            // Optional parameters for step 1 of APAtrap: identify distal 3'utr
+
+            // Window size used to scan the mapping result, default is 100.
+            w = 100
+            // Pre-extension size of each 3'UTR, default is 10000.
+            e = 10000
+            // minimum coverage of the end of the distal 3'UTR with comparing to the whole transcript, default is 0.05.
+            c = 0.05
+            // minimum percentage of valid nucleotides in a scanning-window, default is 0.8.
+            p = 0.8
+        }
+        'predict_apa' {
+            // Optional and necessary parameters for step 2 of APAtrap: predict apa
+
+            // Necessary paramters
+            // number of groups (treatments/conditions) of the input files, e.g. -g 2.
+            g = 1
+            // number of files(biological replicates) in each group (treatment/condition), e.g. -n 1 1.
+            // if there is only one file and one group, set n = 1
+            n = '1'
```

Comment:
Copy pasting the reply on the readme here since the question is around the same issue :D 

Really good point! I clarified with the author and he said that we should put input files by group and in the some order of values specified by parameter -n. So for example, if we have two replicates for each condition, we'll then do

predictAPA -i condition1_replicate1.bedgraph condition1_replicate2.bedgraph condition2_replicate1.bedgraph condition2_replicate2.bedgraph -g 2 -n 2 2 -u hg19.utr.bed -o output.txt

This means that the order of files in the samplesheet has no restrictions but I'll probably have to update the workflow such that it groups sample files by condition and then concatenates them next to each other